lib/dialects/abstract/query-generator.js

```diff
@@ -1713,6 +1984,68 @@ const QueryGenerator = {
     return fragment;
   },
 
+  /*
+     Takes a properly formated CTE attribute definiton from either the intial or
+     recursive step and recursively turns it into the correct SQL snippet.
+   */
+  attributeItemQuery(object, options, modelName, cteName) {
+    options = options || {};
+
+    let operands;
+
+    const operatorMap = {
+      $add: '+',
+      $sub: '-',
```

Comment:
This needs to be Symbols, but I am not sure how to expose them with other operators

may be call them `Op.cteAdd` ?

Reply:
Making special CTE operators sounds like the right approach since they function very differently than the other operators of the same name.